src/components/guide/index.js

```diff
@@ -0,0 +1,190 @@
+
+    let intro = introduction.split(".", 1)[0];
+    intro = `${intro}.`;
+    const readMore = introduction.replace(intro, "").trim();
+    return {intro, readMore};
+  }
+  selectBoxOrTabs() {
+    const {displayedCompetency} = this.state;
+
+    if(smallScreen()) {
+      return (
+        <div>
+          <select
+            value={displayedCompetency.name}
+            className={style.mobileSelect}
+            onChange={(e) => this.displayCompetency(e)}
+          >
+            {this.state.competencies.map((competency) => (
+              <option value={competency.name}>{competency.name}</option>
+            ))
+            }
+          </select>
+          <p className={style.mobileBadge}>
+            <img src={`https://cdn.traitify.com/assets/images/js/${this.tabImage(displayedCompetency.name)}`} alt={`${displayedCompetency.name} badge`} />
+          </p>
+        </div>
+      );
+    } else {
+      return (
+        <ul className={style.tabs}>
+          {this.state.competencies.map((competency, index) => (
+            <li
+              className={competency.name === displayedCompetency.name ? style.tabActive : null}
+              key={competency.id}
+            >
+              <a
+                href={`#tab-${index}`}
+                tabIndex={0}
+                onKeyPress={() => this.displayCompetency(competency.name)}
+                onClick={() => this.displayCompetency(competency.name)}
+                name={competency.name}
+              >
+                <img src={`https://cdn.traitify.com/assets/images/js/${this.tabImage(competency.name)}`} alt={`${competency.name} badge`} />
+                <br />
+                {competency.name}
+              </a>
+            </li>
+          ))}
+        </ul>
+      );
+    }
+  }
+  setGuide() {
+    const params = {assessmentId: this.props.assessmentID};
+    this.props.traitify.graphqlQuery("/interview_guides/graphql", guideQuery({params}))
+      .then((response) => {
+        if(response.errors) { this.setState({errors: response.errors}); return; }
+        const {competencies} = response.data.guide;
+        this.setState({competencies, displayedCompetency: competencies[0]});
+      });
+  }
+  stringToListItems(entity) {
+    let entities = entity.replace("\n", "<br/>").split("<br/>");
+    entities = entities.map((e) => <li>{e}</li>);
+
+    return (
+      <ul>
+        {entities}
+      </ul>
+    );
+  }
+  tabImage(tab) {
+    switch(tab) {
+      case "Solving Problems": return "openness.png";
+      case "Delivering Results": return "conscientiousness.png";
+      case "Engaging with People": return "extraversion.png";
+      case "Influencing People": return "agreeableness.png";
+      case "Managing Pressure": return "emotional_stability.png";
+      default: return null;
+    }
+  }
+  render() {
+    if(this.state.errors.length > 0) { return <div />; }
+    if(this.state.competencies.length === 0) { this.setGuide(); return <div />; }
```

Comment:
This should be moved into a `componentDidMount`, otherwise rerenders from props changes will trigger additional calls while the first request is completing